compositor_render/src/renderer/texture.rs

```diff
@@ -126,8 +72,157 @@ impl Texture {
     }
 }
 
+pub struct YUVTextures(pub [Texture; 3]);
+
+impl YUVTextures {
+    pub fn new(ctx: &WgpuCtx, resolution: &Resolution) -> Self {
+        Self([
+            Self::new_plane(ctx, resolution.width, resolution.height),
+            Self::new_plane(ctx, resolution.width / 2, resolution.height / 2),
+            Self::new_plane(ctx, resolution.width / 2, resolution.height / 2),
+        ])
+    }
+
+    fn new_plane(ctx: &WgpuCtx, width: usize, height: usize) -> Texture {
+        Texture::new(
+            ctx,
+            None,
+            wgpu::Extent3d {
+                width: width as u32,
+                height: height as u32,
+                depth_or_array_layers: 1,
+            },
+            wgpu::TextureFormat::R8Unorm,
+            wgpu::TextureUsages::RENDER_ATTACHMENT
+                | wgpu::TextureUsages::TEXTURE_BINDING
+                | wgpu::TextureUsages::COPY_DST
+                | wgpu::TextureUsages::COPY_SRC
+                | wgpu::TextureUsages::TEXTURE_BINDING,
+        )
+    }
+
+    pub fn new_bind_group_layout(device: &wgpu::Device) -> wgpu::BindGroupLayout {
+        let create_entry = |binding: u32| wgpu::BindGroupLayoutEntry {
+            binding,
+            ty: Texture::DEFAULT_BINDING_TYPE,
+            visibility: wgpu::ShaderStages::FRAGMENT,
+            count: None,
+        };
+        device.create_bind_group_layout(&wgpu::BindGroupLayoutDescriptor {
+            label: Some("yuv all textures bind group layout"),
+            entries: &[create_entry(0), create_entry(1), create_entry(2)],
+        })
+    }
+
+    pub(self) fn new_bind_group(
```

Comment:
```suggestion
    fn new_bind_group(
```
`pub(self)` is equivalent to no `pub` at all, I believe

Reply:
it makes sense if you have more than one module in the same file, for some reason I assumed that you need it to access private fields in the same file, but this does not seem to be necessary